packages/graphql-auth-transformer/src/ModelAuthTransformer.ts

```diff
@@ -236,24 +315,27 @@ Static group authorization should perform as expected.`
         const isDeleteRule = isOpRule('delete');
         // The field handler adds the read rule on the object
         const readRules = rules.filter((rule: AuthRule) => isReadRule(rule))
-        this.protectField(ctx, parent.name.value, definition.name.value, readRules)
+        this.protectField(ctx, parent.name.value, definition,
+            readRules, protectPrivateFields)
 
         // Protect mutations when objects including this field are trying to be created.
         const createRules = rules.filter((rule: AuthRule) => isCreateRule(rule))
         this.protectCreateForField(ctx, parent, definition, createRules)
 
         // Protect update mutations when objects inluding this field are trying to be updated.
         const updateRules = rules.filter((rule: AuthRule) => isUpdateRule(rule))
-        this.protectUpdateForField(ctx, parent, definition, updateRules)
+        this.protectUpdateForField(ctx, parent, definition,
+            updateRules)
 
         // Delete operations are only protected by @auth directives on objects.
         const deleteRules = rules.filter((rule: AuthRule) => isDeleteRule(rule))
         this.protectDeleteForField(ctx, parent, definition, deleteRules)
     }
 
-    private protectField(ctx: TransformerContext, typeName: string, fieldName: string, rules: AuthRule[]) {
+    private protectField(ctx: TransformerContext, typeName: string,
+        field: FieldDefinitionNode, rules: AuthRule[], protectPrivateFields?: boolean) {
```

Comment:
Why is protectPrivateFields optional? make it mandatory with a default value, then code would be more explicit inside IMHO.